### PR TITLE
Add support for validating cloudantDatabase

### PR DESCRIPTION
**Pull request by alexsm82**
Add support for validating `cloudantDatabase` elements.

### CODE REVIEW

File: dev/com.ibm.ws.rest.handler.validator.cloudant/src/com/ibm/ws/rest/handler/validator/cloudant/CloudantDatabaseValidator.java [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+        LinkedHashMap<String, Object> result = new LinkedHashMap<String, Object>();
+
+        try {
+            ResourceConfig config = null;
+            int authType = "container".equals(auth) ? 0 : "application".equals(auth) ? 1 : -1;
+            if (authType >= 0) {
+                config = resourceConfigFactory.createResourceConfig("com.cloudant.client.api.Database");
+                config.setResAuthType(authType);
+                if (authAlias != null)
+                    config.addLoginProperty("DefaultPrincipalMapping", authAlias); // set provided auth alias
```
Comment (njr-11) [inline dev/com.ibm.ws.rest.handler.validator.cloudant/src/com/ibm/ws/rest/handler/validator/cloudant/CloudantDatabaseValidator.java]: Auth alias shouldn't be added if application auth

File: dev/com.ibm.ws.rest.handler.validator.cloudant/src/com/ibm/ws/rest/handler/validator/cloudant/CloudantDatabaseValidator.java [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+            }
+
+            Object database = ((ResourceFactory) instance).createResource(config);
+            //There isn't anything particularly useful in the DB Info, but invoking the method
+            //ensures that the database exists (or it will be created if create="true")
+            database.getClass().getMethod("info").invoke(database);
```
Comment (njr-11) [inline dev/com.ibm.ws.rest.handler.validator.cloudant/src/com/ibm/ws/rest/handler/validator/cloudant/CloudantDatabaseValidator.java]: let's at least include the Cloudant driver version, and should check if we can read anything from the db about the db version info

File: dev/com.ibm.ws.rest.handler.validator.cloudant/src/com/ibm/ws/rest/handler/validator/cloudant/CloudantDatabaseValidator.java (hunk 1 above)
Comment (alexsm82) [reply dev/com.ibm.ws.rest.handler.validator.cloudant/src/com/ibm/ws/rest/handler/validator/cloudant/CloudantDatabaseValidator.java]: Yep, although the underlying cloudant container code correctly handles the scenario I'll update it.

**Comment by alexsm82** [comment]
#build

**Comment by alexsm82** [comment]
#build

**Comment by alexsm82** [comment]
#build

**Comment by alexsm82** [comment]
The build had a single failure, which was an unrelated, known issue in the mp config unit tests.

**Pull request merged**